### ISSUE TITLE
Can't cast database type .<unknown> to Vector

### ISSUE THREAD

**Issue by Sharkovskyi** [pgvector/pgvector-dotnet#39]
Hello, the application works on the 6th dotnet with the entity framework. I recently discovered a bug that occurs during the first migration with the creation of a database, and throws an Invalid Cast Exception when receiving data. Restarting the application helps, and provided that the migrations have already been applied to the database (namely, the first creation, etc.), then everything starts to work well.

DBContext:
```c#
public class AICommunicationContext : DbContext
{
    public DbSet<AIPrompt> AIPrompts { get; set; } = null!;
    public DbSet<Document> Documents { get; set; } = null!;
    public DbSet<Embedding> Embeddings { get; set; } = null!;
    public DbSet<DocumentSharing> DocumentSharing { get; set; } = null!;

    public AICommunicationContext() { }
    public AICommunicationContext(DbContextOptions<AICommunicationContext> options) : base(options) { }

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        if (!optionsBuilder.IsConfigured)
        {
            var connectionString = EnvironmentVariables.GetConnectionStringAsync.GetAwaiter().GetResult()
                .Replace(ConfigConstants.DbReplacePattern, ConfigHelper.AppSetting(ConfigConstants.DbName));
            optionsBuilder.UseNpgsql(connectionString, o => o.UseVector());
        }
    }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.HasPostgresExtension("vector");
        modelBuilder.ApplyConfigurationsFromAssembly(typeof(AICommunicationContext).Assembly);
    }
}
```
Service registration:
```c#
.AddDbContext<AICommunicationContext>(c => c.UseNpgsql(connectionString, o => o.UseVector()))
```

MigrateDbMethod:
```c#
static async Task MigrateDatabaseAsync(IHost app)
{
    using var scope = app.Services.CreateScope();
    var services = scope.ServiceProvider;

    var context = services.GetRequiredService<AICommunicationContext>();
    if ((await context.Database.GetPendingMigrationsAsync()).Any())
    {
        await context.Database.MigrateAsync();
    }
}
```

The method which returns an error:
```c#
    public async Task<List<Embedding>> GetTheClosestEmbeddingAsync(Guid userId, string fileName, Vector vector)
    {
        return await _context.Embeddings
            .Where(a => a.Document.Name.Equals(fileName) && (a.Document.UserId == userId || a.Document.DocumentSharing
                .Where(a => a.Document.Name.Equals(fileName)).Any(a => a.UserId == userId)))
            .OrderBy(a => a.Vector!.L2Distance(vector))
            .Take(2)
            .ToListAsync();
    }
```

![image](https://github.com/pgvector/pgvector-dotnet/assets/140795985/60f9712a-8742-4384-882b-fd4ca0869b02)

I repeat this happens only at the first initialization of the application, provided that the database needs to be raised from 0, when restarting when the base is raised, everything is fine

**Comment by roji** [comment]
> I repeat this happens only at the first initialization of the application

See https://github.com/npgsql/efcore.pg/issues/292 - this has been fixed in version 8.0 of the Npgsql provider. [Here's](https://github.com/npgsql/efcore.pg/issues/292) a workaround for older verisons.

**Comment by ankane** [comment]
Thanks @roji

**Comment by Sharkovskyi** [comment]
> > I repeat this happens only at the first initialization of the application
> 
> See [npgsql/efcore.pg#292](https://github.com/npgsql/efcore.pg/issues/292) - this has been fixed in version 8.0 of the Npgsql provider. [Here's](https://github.com/npgsql/efcore.pg/issues/292) a workaround for older versions.

Thank you very much for your help!
